android/java/org/chromium/chrome/browser/playlist/kotlin/activity/PlaylistActivity.kt

```diff
@@ -0,0 +1,554 @@
+                    mTvPlaylistTotalSize.text =
+                        Formatter.formatShortFileSize(this@PlaylistActivity, totalFileSize)
+                }
+
+                Glide.with(this@PlaylistActivity)
+                    .load(playlist.items[0].thumbnailPath.url)
+                    .placeholder(R.drawable.ic_playlist_placeholder)
+                    .error(R.drawable.ic_playlist_placeholder)
+                    .into(mIvPlaylistCover)
+
+                mTvTotalMediaCount.text =
+                    getString(R.string.playlist_number_items, mPlaylist.items.size)
+                mTvPlaylistName.text =
+                    if (mPlaylist.id == ConstantUtils.DEFAULT_PLAYLIST)
+                        resources.getString(R.string.playlist_play_later)
+                    else mPlaylist.name
+
+                mPlaylistItemAdapter =
+                    PlaylistItemAdapter(this@PlaylistActivity, this@PlaylistActivity)
+                mPlaylistItemAdapter?.submitList(playlist.items.toMutableList())
+                mPlaylistItemAdapter?.let {
+                    val callback =
+                        PlaylistItemGestureHelper(
+                            this@PlaylistActivity,
+                            mRvPlaylist,
+                            it,
+                            this@PlaylistActivity
+                        )
+                    mItemTouchHelper = ItemTouchHelper(callback)
+                    mItemTouchHelper.attachToRecyclerView(mRvPlaylist)
+                }
+                mRvPlaylist.adapter = mPlaylistItemAdapter
+                mPlaylistItemAdapter?.setEditMode(false)
+                mPlaylistToolbar.enableEditMode(false)
+                mIvPlaylistOptions.visibility = View.VISIBLE
+                mPlaylistView.visibility = View.VISIBLE
+                mEmptyView.visibility = View.GONE
+
+                mProgressBar.visibility = View.GONE
+                mRvPlaylist.visibility = View.VISIBLE
+            } else {
+                setEmptyView()
+            }
+        }
+    }
+
+    private fun setEmptyView() {
+        mIvPlaylistCover.setImageResource(R.drawable.ic_playlist_placeholder)
+        mEmptyView.visibility = View.VISIBLE
+        mPlaylistView.visibility = View.GONE
+    }
+
+    private fun stopVideoPlayerOnDelete(selectedPlaylistItem: PlaylistItem) {
+        mMediaBrowser?.currentMediaItem?.mediaId?.let {
+            if (it == selectedPlaylistItem.id) {
+                mMediaBrowser?.stop()
+            }
+        }
+    }
+
+    private fun openPlaylistPlayer(isShuffle: Boolean, position: Int) {
+        val browser = this.mMediaBrowser ?: return
+
+        var recentPlaylistIds = LinkedList<String>()
+        val recentPlaylistJson =
+            PlaylistPreferenceUtils.defaultPrefs(this@PlaylistActivity).recentlyPlayedPlaylist
+        if (!recentPlaylistJson.isNullOrEmpty()) {
+            recentPlaylistIds =
+                GsonBuilder()

```

Comment:
the same question as above about json

Reply:
@SergeyZhukovsky we are using coroutines here. i am updating it with chromium recommended posttask